crates/partition-store/src/invocation_status_table/mod.rs

```diff
@@ -212,6 +220,84 @@ impl ScanInvocationStatusTable for PartitionStore {
         )
         .map_err(|_| StorageError::OperationalError)
     }
+
+    fn for_each_invocation_status<
+        F: FnMut((InvocationId, InvocationStatus)) -> ControlFlow<()> + Send + Sync + 'static,
+    >(
+        &self,
+        range: RangeInclusive<PartitionKey>,
+        f: F,
+    ) -> Result<impl Future<Output = Result<()>> + Send> {
+        // these two iterators can run concurrently in theory.
+        // in practice, there are not typically any keys in the first iterator, but rust rightfully forces us to use a mutex to protect the FnMut
+        // the intent here is that iterators race to produce the first row, the winner gets an owned lock guard which it then holds until its done iterating
+        // and then the next iterator is able to get a lock guard.
+
+        let f = Arc::new(tokio::sync::Mutex::new(f));
+
+        let old_status_keys = self
+            .iterator_for_each(
+                "df-for-each-invocation-status-v1",
+                Priority::Low,
+                TableScan::FullScanPartitionKeyRange::<InvocationStatusKeyV1>(range.clone()),
+                {
+                    let f = f.clone();
+                    let mut f_locked = None;
+                    move |(mut key, mut value)| {
+                        let state_key =
+                            break_on_err(InvocationStatusKeyV1::deserialize_from(&mut key))?;
+                        let state_value = break_on_err(InvocationStatusV1::decode(&mut value))?;
+
+                        let (partition_key, invocation_uuid) =
+                            break_on_err(state_key.into_inner_ok_or())?;
+
+                        let f = f_locked.get_or_insert_with(|| f.clone().blocking_lock_owned());
+
+                        f((
+                            InvocationId::from_parts(partition_key, invocation_uuid),
+                            state_value.0,
+                        ))
+                        .map_break(Ok)
+                    }
+                },
+            )
+            .map_err(|_| StorageError::OperationalError)?;
```

Comment:
You don't need this anymore in 1.5, because we migrate all the old v1 on boot.

Reply:
should we remove that then?